test-framework/sudo-compliance-tests/src/sudoers/env_keep.rs

```diff
@@ -0,0 +1,387 @@
+use sudo_test::{Command, Env};
+
+use crate::{helpers, Result, SUDOERS_ALL_ALL_NOPASSWD};
+
+#[test]
+fn equal_single() -> Result<()> {
+    let env_name = "SHOULD_BE_PRESERVED";
+    let env_val = "42";
+    let env = Env([
+        SUDOERS_ALL_ALL_NOPASSWD,
+        &*format!("Defaults env_keep = {env_name}"),
+    ])
+    .build()?;
+
+    let stdout = Command::new("env")
+        .arg(format!("{env_name}={env_val}"))
+        .args(["sudo", "env"])
+        .exec(&env)?
+        .stdout()?;
+    let sudo_env = helpers::parse_env_output(&stdout)?;
+
+    assert_eq!(Some(env_val), sudo_env.get(env_name).copied());
+
+    Ok(())
+}
+
+#[test]
+fn equal_multiple() -> Result<()> {
+    let env_name1 = "SHOULD_BE_PRESERVED";
+    let env_name2 = "ALSO_SHOULD_BE_PRESERVED";
+    let env_val1 = "42";
+    let env_val2 = "24";
+    let env = Env([
+        SUDOERS_ALL_ALL_NOPASSWD,
+        &*format!("Defaults env_keep = \"{env_name1} {env_name2}\""),
+    ])
+    .build()?;
+
+    let stdout = Command::new("env")
+        .arg(format!("{env_name1}={env_val1}"))
+        .arg(format!("{env_name2}={env_val2}"))
+        .args(["sudo", "env"])
+        .exec(&env)?
+        .stdout()?;
+    let sudo_env = helpers::parse_env_output(&stdout)?;
+
+    assert_eq!(Some(env_val1), sudo_env.get(env_name1).copied());
+    assert_eq!(Some(env_val2), sudo_env.get(env_name2).copied());
+
+    Ok(())
+}
+
+#[test]
+fn equal_repeated() -> Result<()> {
+    let env_name = "SHOULD_BE_PRESERVED";
+    let env_val = "42";
+    let env = Env([
+        SUDOERS_ALL_ALL_NOPASSWD,
+        &*format!("Defaults env_keep = \"{env_name} {env_name}\""),
+    ])
+    .build()?;
+
+    let stdout = Command::new("env")
+        .arg(format!("{env_name}={env_val}"))
+        .args(["sudo", "env"])
+        .exec(&env)?
+        .stdout()?;
+    let sudo_env = helpers::parse_env_output(&stdout)?;
+
+    assert_eq!(Some(env_val), sudo_env.get(env_name).copied());
+
+    Ok(())
+}
+
+#[test]
+fn equal_overrides() -> Result<()> {
+    let env_name1 = "SHOULD_BE_PRESERVED";
+    let env_val1 = "42";
+    let env_name2 = "SHOULD_BE_REMOVED";
+    let env_val2 = "24";
+    let env = Env([
+        SUDOERS_ALL_ALL_NOPASSWD,
+        &*format!("Defaults env_keep = \"{env_name1} {env_name2}\""),
+        &*format!("Defaults env_keep = {env_name1}"),
+    ])
+    .build()?;
+
+    let stdout = Command::new("env")
+        .arg(format!("{env_name1}={env_val1}"))
+        .arg(format!("{env_name2}={env_val2}"))
+        .args(["sudo", "env"])
+        .exec(&env)?
+        .stdout()?;
+    let sudo_env = helpers::parse_env_output(&stdout)?;
+
+    assert!(sudo_env.get(env_name2).is_none());
```

Comment:
shouldn't we check that `env_name1` has the value we provided? I know we already tested this indirectly in other tests but even then.

Reply:
that's a good call. I'll add another assertion